### DISCUSSION TITLE
Uninstalling a formula should disable and remove its service

### DISCUSSION THREAD

**Discussion by rolandcrosby-columntax** [Homebrew/discussions#4659] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.29
ORIGIN: https://github.com/Homebrew/brew
HEAD: a2a05f96ef6215643f599e7bfeaacd4f34e7cd9a
Last commit: 31 hours ago
Core tap JSON: 19 Jul 16:20 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_blizzard_avalanche
Clang: 14.0.3 build 1403
Git: 2.40.1 => /opt/homebrew/bin/git
Curl: 7.88.1 => /usr/bin/curl
macOS: 13.4.1-arm64
CLT: 14.3.1.0.1.1683849156
Xcode: 14.3.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I used to have Postgres 13 installed on my Mac (`brew install postgresql@13`) and running automatically at login (`brew services start postgresql@13`). I recently upgraded to a new version and uninstalled Postgres 13 (`brew uninstall postgresql@13`), and was surprised to see that a `homebrew.mxcl.postgresql@13.plist` service still existed in `~/Library/LaunchAgents`, which would try to start the Postgres 13 server in the now-gone path `/opt/homebrew/opt/postgresql@13`. 

I would expect `brew uninstall` to detect that the formula being uninstalled provides a service, check if the service is enabled, and remove the launch agent configuration if one exists. It would be nice if this also elegantly handled the case when a service was installed globally via `sudo brew services start`.

**Comment by karimbenbourenane** [comment]
Almost two years later and this still hasn't been implemented? It's still an actual problem. Install any service with a launch agent plist and when uninstalling it just leaves the plist add to cruft in `~/Library/LaunchAgents`. I can't see any possible reason to leave it there especially when the plists are in the form `homebrew.mxcl.<formula>.plist` such as `homebrew.mxcl.redis.plist` or `homebrew.mxcl.postgresql@17.plist`, these are clearly associated with a homebrew formula and wouldn't be there from a manual install using an official installer.

**Reply by SMillerDev** [reply]
It appears nobody found it as important and you did and volunteered their time to fix the problem. But I'll happily review your pull request when it comes around.
